src/transformers/generation_logits_process.py

```diff
@@ -689,3 +689,72 @@ class LogitNormalization(LogitsProcessor, LogitsWarper):
     def __call__(self, input_ids: torch.Tensor, scores: torch.Tensor) -> torch.Tensor:
         scores = scores.log_softmax(dim=-1)
         return scores
+
+
+class SuppressBlank(LogitsProcessor):
+    r""" """
+
+    def __init__(self, tokenizer, sample_begin: int = 1):
+        self.tokenizer = tokenizer
```

Comment:
Let's not pass the tokenizer here, but instead pass the two tokens we need below (token for `" "` and the `eos` or `eot` token)

Reply:
Fixed !